### PR TITLE
Cleanup poms

### PR DESCRIPTION
**Pull request by realityforge**
Cleans up poms by removing reference to deprecated parent POM and by removing dependencies that are transitively available.

### CODE REVIEW

File: maven/pom-dom.xml [hunk 1]
```diff
@@ -19,16 +13,6 @@
...
-    <dependency>
-      <groupId>com.google.jsinterop</groupId>
-      <artifactId>jsinterop-annotations</artifactId>
-      <version>1.0.2</version>
-    </dependency>
-    <dependency>
-      <groupId>com.google.jsinterop</groupId>
-      <artifactId>base</artifactId>
-      <version>1.0.0-RC1</version>
-    </dependency>
```
[the remark this answers is not in the document]
Comment (tbroyer) [reply maven/pom-dom.xml]: Maven best practices (per the maven-dependency-plugin) is to include everything that you directly depend on. Most common practice is to just rely on whatever is on your compile classpath.
Maven is missing an implementation vs api dependencies (as now exists in Gradle).

So I agree those dependencies should be kept here.

[the remark this answers is not in the document]
Comment (realityforge) [reply maven/pom-dom.xml]: Okay removed the commit that made these changes to the pom

**Comment by realityforge** [comment]
A friendly ping! ;)

**Comment by jDramaix** [comment]
Thanks for the contribution

**Pull request closed without merging**